src/components/ec-modal/ec-modal.spec.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-use-before-define */
+import { mount } from '@vue/test-utils';
+import EcModal from './ec-modal.vue';
+
+describe('EcModal', () => {
+  it('should not render the modal', () => {
+    const wrapper = mount(EcModal);
+    expect(wrapper.find('.ec-modal').exists()).toBe(false);
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should render basic modal', () => {
+    const wrapper = mount(EcModal, {
+      propsData: {
+        showModal: true,
+      },
+      slots: {
+        main: '<p>Before we can process your application we need you to upload your management accounts. You can do this now or leave it for later.</p>',
+      },
+    });
+    expect(wrapper.element).toMatchSnapshot();
+  });
+  it('should render close button', () => {
+    const wrapper = mount(EcModal, {
```

Comment:
To make the tests repeating less code we introduced a function for mounting components:
```
function mountModal(props, mountOpts) {
  return mount(EcModal, {
    propsData: { ...props },
    ...mountOpts,
  });
}
```
```
it('should do something', () => {
  const wrapper = mountModal({ showModal: true }); // shorter and simpler
});
```

You can add more defaults for the components here, mocks, stubs, store, router, default props, etc.